src-common/main/scala/WSClient.scala

```diff
@@ -28,21 +28,23 @@ package object audit {
   // these are for internal use only, but can't be package private since clients are in uk.gov.hmrc.audit and uk.gov.hmrc.play.audit...
   object WSClient {
     def apply(
-      connectTimeout: Duration,
-      requestTimeout: Duration,
-      userAgent     : String
-    )(implicit
-      materializer: Materializer
-    ): WSClient =
+               connectTimeout: Duration,
+               requestTimeout: Duration,
+               userAgent     : String
+             )(implicit
+               materializer: Materializer
+             ): WSClient =
       StandaloneAhcWSClient(
         config = AhcWSClientConfigFactory.forConfig()
-                   .copy(wsClientConfig = WSClientConfig()
-                     .copy(
-                       connectionTimeout = connectTimeout,
-                       requestTimeout    = requestTimeout,
-                       userAgent         = Some(userAgent)
-                     )
-                   )
+          .copy(
+            wsClientConfig = WSClientConfig()
+              .copy(
+                connectionTimeout = connectTimeout,
+                requestTimeout    = requestTimeout,
+                userAgent         = Some(userAgent),
+              ),
+            maxRequestRetry   = 0
```

Comment:
Is this relying on envoy retrying 5 times? If envoy behaviour was changed, would this be acceptable to never retry?
The worst case of 25 retries be default would only occur for consistent IOExceptions, if this is unlikely, aren't the defaults acceptable?

Reply:
I'm on the fence about this one, to be honest.  

Our standard 'ask' is 2 attempts; one to datastream, the other to the secondary channel (logging in this case).  Retries to help get the message through, though.  

It complicates testing (manual and automatic) a little to include it, and I'm not sure how much safety it buys us.  For instance, if you can't establish a connection to envoy, an extra 5 attempts might not help envoy get back on its feet. 

In SUP-12021, it seems that envoy may be OOMing with its memory footprint increasing the number of requests concurrent open (not yet conclusive, but some signs are there).  If a high-volume service is already overwhelming/on the verge of overwhelming envoy with its throughput, adding 4 additional concurrent requests seems more likely to make it worse than help. 

We saw some similar (again, not conclusive) indications in DLK-852 - where high volume (concurrency?) of requests coming from one instance was correlated with some troubles establishing connections to datastream.